web/src/main/webapp/my-app/layout/controllers.js

```diff
@@ -47,11 +47,127 @@ define(['angular', 'jquery'], function(angular, $) {
     function($localStorage, $log, $sessionStorage,
       $scope, $rootScope, layoutService) {
       var vm = this;
+
+      /**
+       * Log whenever a widget is moved
+       * @param eventType {String} Kind of interaction that moved the widget
+       * @param fname {String} The moved widget's fname
+       * @param dropIndex {Number} Index where the widget landed
+       * @param startIndex {Number} (optional) Index before moving the widget
+       * @return {boolean}
+       */
+      $scope.logEvent = function(eventType, fname, dropIndex, startIndex) {
+        switch (eventType) {
+          case 'dragEnd':
+            $log.info('Dragged ' + fname + ' to index ' + dropIndex);
+            break;
+          case 'keyboardMove':
+            $log.info('Moved ' + fname + 'from index ' + startIndex +
+              ' to index ' + dropIndex);
+            break;
+          default:
+            return true;
+        }
+      };
+
+      /**
+       * Respond to arrow key-presses when focusing a movable list element
+       * @param widget {Object} The widget trying to move
+       * @param event {Object} The event object
+       * @return {boolean} True if proper conditions aren't met
+       */
+      $scope.moveWithKeyboard = function(widget, event) {
```

Comment:
Could something like fluid drag and drop work here? https://wiki.fluidproject.org/display/fluid/Drag+and+Drop+Design+Pattern

Or could this be contributed upstream so that everyone can enjoy an accessible experience?

Reply:
Personally I don't agree that sortable widgets is something that should be in the app framework. I don't even think widgets should be in the app framework at all. We included them because on-campus developers using the framework to develop other apps asked us to, and in retrospect I would have fought against doing it. 

Hopefully there comes a day when the app framework becomes a collection of API-driven web components. Then this separation of concerns won't be an issue. 🤷‍♂️

Reply:
https://github.com/marceljuenemann/angular-drag-and-drop-lists/issues/348

Reply:
> I don't agree that sortable widgets is something that should be in the app framework.

I agree :+1:
Hence the suggestion to contribute it upstream at Angular drag and drop lists.
Thanks @apetro for providing the link to the issue. :bow: